app/src/main/java/com/poklad/androidifystore/presentation/ui/screens/all_products/AllProductsFragment.kt

```diff
@@ -0,0 +1,103 @@
+import com.poklad.androidifystore.utils.Resource
+import com.poklad.androidifystore.utils.invisible
+import com.poklad.androidifystore.utils.log
+import com.poklad.androidifystore.utils.visible
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class AllProductsFragment : BaseFragment<FragmentAllProductsBinding, BaseViewModel>() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    override val viewModel: AllProductsViewModel by viewModels {
+        viewModelFactory
+    }
+    private val allProductsAdapter: AllProductsAdapter by lazy {
+        AllProductsAdapter()
+    }
+
+    override fun inflateViewBinding(inflater: LayoutInflater): FragmentAllProductsBinding =
+        FragmentAllProductsBinding.inflate(inflater)
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        StoreApp.daggerComponent.inject(this)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initRecyclerView()
+        setUpObserver()
+    }
+
+    private fun setUpObserver() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.products.collect { resource ->
+                    when (resource) {
+                        is Resource.Success -> {
+                            binding.apply {
+                                progressBarAllProducts.invisible()
+                                renderList(resource.data)
+                                recycleViewProductList.visible()
+                            }
+                        }
+
+                        is Resource.Loading -> {
+                            binding.apply {
+                                progressBarAllProducts.visible()
+                                recycleViewProductList.invisible()
+                            }
+                        }
+
+                        is Resource.Error -> {
+                            binding.progressBarAllProducts.visible()
+                            Toast.makeText(
+                                requireContext(),
+                                resource.throwable.toString(),
+                                Toast.LENGTH_LONG
+                            ).show()
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
+    private fun renderList(productsList: List<ProductItem>) {
+        allProductsAdapter.list = productsList
+    }
+
+    private fun initRecyclerView() {
+        binding.recycleViewProductList.apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            adapter = allProductsAdapter
+        }
+        allProductsAdapter.setOnclickListener {
+            findNavController().navigate(R.id.action_allProductsFragment_to_productDetailsFragment)
```

Comment:
please create `navigateTo` method with two arguments (destination and parameters: Pair<>) in your `BaseFragment` and use it

Reply:
done